catroid/src/org/catrobat/catroid/ui/adapter/LookAdapter.java

```diff
@@ -59,7 +60,14 @@ public long getItemId(int position) {
 			return INVALID_ID;
 		}
 		LookData item = getItem(position);
-		return idMap.get(item);
+
+		for (Map.Entry<LookData, Integer> e : idMap.entrySet()) {
+			if (item.equals(e.getKey())) {
+				return e.getValue();
+			}
+		}
+
+		return INVALID_ID;
```

Comment:
Please fix `LookData.hashCode` and `LookData.equals` instead. Both methods should also work if name or fileName are null.

---

The hashCode calculation is also wrong (`+ super.hashCode()`). But I'm not sure if we should fix that as well before the release. There may be side effects I'm not aware of. @thmq, what do you think?